Script/PRONTO.py

```diff
@@ -826,29 +965,46 @@ def remisse_mail_writer(remisse_file,ipd_no,ipd_consent,DNA_normal_sampleID,RNA_
 	doc.save(remisse_file)
 
 
-def update_clinical_master_file(InPreD_clinical_data_file,sample_id,if_generate_report,ipd_birth_year,clinical_diagnosis,ipd_gender,ipd_consent,material_id,ipd_collection_year,requisition_hospital,extraction_hospital,tumor_content_nr,batch_nr):
+def update_clinical_master_file(InPreD_clinical_data_file,sample_id,if_generate_report,ipd_birth_year,clinical_diagnosis,ipd_gender,ipd_consent,material_id,ipd_collection_year,requisition_hospital,extraction_hospital,tumor_content_nr,batch_nr,encoding_sys):
 	global ipd_diagnosis_year
 	global runID
 	if_exist = False
 	new_content = ""
-	with open(InPreD_clinical_data_file, 'r', encoding="ISO-8859-1") as fr:
-		for ln in fr:
-			if(ln.split('\t')[0] == sample_id):
-				if_exist = True
-				line = sample_id + "\t" + runID + "\t" + if_generate_report + "\t" + ipd_birth_year + "\t" + ipd_diagnosis_year + "\t" + clinical_diagnosis + "\t" + ipd_gender[0] + "\t" + ipd_consent + "\t" + material_id + "\t" + ipd_collection_year + "\t" + requisition_hospital + "\t" + extraction_hospital + "\t" + str(tumor_content_nr) + "\t" +batch_nr + "\n"
-				new_line = ln.replace(ln,line)
-				new_content = new_content + new_line
-			else:
-				new_content = new_content + ln
+	if(encoding_sys != ""):
+		fr = open(InPreD_clinical_data_file, 'r', encoding=encoding_sys)
+	else:
+		fr = open(InPreD_clinical_data_file, 'r')
+	if(str(requisition_hospital) == "0.0"):
+		requisition_hospital = "-"
+	if(str(extraction_hospital) == "0.0"):
+		extraction_hospital = "-"
+	for ln in fr:
+		if(ln.split('\t')[0] == sample_id):
+			if_exist = True
+			line = sample_id + "\t" + runID + "\t" + if_generate_report + "\t" + ipd_birth_year + "\t" + ipd_diagnosis_year + "\t" + clinical_diagnosis + "\t" + ipd_gender[0] + "\t" + ipd_consent + "\t" + material_id + "\t" + ipd_collection_year + "\t" + requisition_hospital + "\t" + extraction_hospital + "\t" + str(tumor_content_nr) + "\t" + batch_nr + "\n"
+			new_line = ln.replace(ln,line)
+			new_content = new_content + new_line
+		else:
+			new_content = new_content + ln
 	fr.close()
+	if(encoding_sys != ""):
+		fa = open(InPreD_clinical_data_file, 'a', encoding=encoding_sys)
```

Comment:
do I understand this correctly, you open the file, do something with the content, close it and open it again? Maybe this can be done in one go. 🤔

Reply:
I am not too sure about python file manipulations but at least you could have a function that allows you to open files easily. Something like:
```python
def openFile(path, permissions, encoding):
  if (encoding != ""):
    return open(path, permission, encoding=encoding)
  else:
    return open(path, permission)'
```

Reply:
I will also create an issue to update the codes next time. :)